store/cache/cache.go

```diff
@@ -106,22 +106,24 @@ func (ckv *CommitKVStoreCache) CacheWrap(storeKey types.StoreKey) types.CacheWra
 // If the value doesn't exist in the write-through cache, the query is delegated
 // to the underlying CommitKVStore.
 func (ckv *CommitKVStoreCache) Get(key []byte) []byte {
-	ckv.mtx.Lock()
-	defer ckv.mtx.Unlock()
+	ckv.mtx.RLock()
 
 	types.AssertValidKey(key)
 
 	keyStr := string(key)
 	value, ok := ckv.cache.Get(keyStr)
 	if ok {
 		// cache hit
+		ckv.mtx.RUnlock()
 		return value
 	}
 
```

Comment:
We need to unlock the readlock before acquire write lock here right?

Reply:
yes good call